querydsl-sql/src/main/java/com/querydsl/sql/types/AbstractJodaTimeDateTimeType.java

```diff
@@ -0,0 +1,35 @@
+package com.querydsl.sql.types;
+
+import org.joda.time.format.DateTimeFormat;
+import org.joda.time.format.DateTimeFormatter;
+
+import java.util.Calendar;
+import java.util.TimeZone;
+
+/**
+ * Common abstract superclass for Type implementations for Joda-Time
+ *
+ * @param <T>
+ */
+public abstract class AbstractJodaTimeDateTimeType<T> extends AbstractType<T> {
```

Comment:
Why did you opt to create a new `AbstractJodaTimeDateTimeType` rather than modify the existing `AbstractDateTimeType` - which was, and still is, dependent on Joda date/time?

The change of signature for the DateTimeType implementation leads to a backwards compatiblity issue:

```
Class com.querydsl.sql.types.AbstractJodaTimeDateTimeType
4323
 Added Class , access abstract public super synchronized
4324
Class com.querydsl.sql.types.DateTimeType
4325
 Changed Class , access 
4326
Class com.querydsl.sql.types.LocalDateTimeType
4327
 Changed Class , access 
4328
Class com.querydsl.sql.types.LocalDateType
4329
 Changed Class , access 
4330
Class com.querydsl.sql.types.LocalTimeType
4331
 Changed Class , access 
4332
[WARNING] Rule 0: org.semver.enforcer.RequireBackwardCompatibility failed with message:
```

Reply:
I modified [AbstractDateTimeType](https://github.com/querydsl/querydsl/pull/2643/files#diff-9255a775bc0f53143b864924632350eb) to no longer depends on Joda time. I created `AbstractJodaTimeDateTimeType` to replace `AbstractDateTimeType` as the parent class for the Joda time converters.

The change of signature for `DateTimeType` is not actually a breaking change, as `DateTimeType` now extends `AbstractJodaTimeDateTimeType`, which itself also extends `AbstractDateTimeType`. Even though the semver enforcer plugin flags it as a breaking change, we will need to increment the major version anyways as this PR itself is a breaking change (users that use Joda time will have to explicitly declare it as a dependency).